packages/contract-utils/ownable/src/ownable.rs

```diff
@@ -0,0 +1,153 @@
+use soroban_sdk::{contracterror, Address, Env, Symbol};
+
+/// A trait for managing contract ownership using a 2-step transfer pattern.
+///
+/// Provides functions to query ownership, initiate a transfer, or renounce
+/// ownership.
+pub trait Ownable {
+    /// Returns `Some(Address)` if ownership is set, or `None` if ownership has
+    /// been renounced.
+    ///
+    /// # Arguments
+    ///
+    /// * `e` - Access to the Soroban environment.
+    fn get_owner(e: &Env) -> Option<Address> {
+        crate::get_owner(e)
+    }
+
+    /// Initiates a 2-step ownership transfer to a new address.
+    ///
+    /// Requires authorization from the current owner. The new owner must later
+    /// call `accept_ownership()` to complete the transfer.
+    ///
+    /// # Arguments
+    ///
+    /// * `e` - Access to the Soroban environment.
+    /// * `new_owner` - The proposed new owner.
+    /// * `live_until_ledger` - Ledger number until which the new owner can
+    ///   accept. A value of `0` cancels any pending transfer.
+    ///
+    /// # Errors
+    ///
+    /// * [`OwnableError::NotAuthorized`] - If the authorization from the
+    ///   current owner is missing.
+    /// * [`stellar_role_transfer::RoleTransferError::NoPendingTransfer`] - If
+    ///   trying to cancel a transfer that doesn't exist.
+    /// * [`stellar_role_transfer::RoleTransferError::InvalidLiveUntilLedger`] -
+    ///   If the specified ledger is in the past.
+    /// * [`stellar_role_transfer::RoleTransferError::InvalidPendingAccount`] -
+    ///   If the specified pending account is not the same as the provided `new`
+    ///   address.
+    ///
+    /// # Notes
+    ///
+    /// * Authorization for the current owner is required.
+    fn transfer_ownership(e: &Env, new_owner: Address, live_until_ledger: u32);
+
+    /// Accepts a pending ownership transfer.
+    ///
+    /// # Arguments
+    ///
+    /// * `e` - Access to the Soroban environment.
+    /// * `caller` - The address of the pending owner accepting ownership.
+    ///
+    /// # Errors
+    ///
+    /// * [`stellar_role_transfer::RoleTransferError::NoPendingTransfer`] - If
+    ///   there is no pending transfer to accept.
+    /// * [`stellar_role_transfer::RoleTransferError::Unauthorized`] - If the
+    ///   caller is not the pending role holder.
+    ///
+    /// # Events
+    ///
+    /// * topics - `["ownership_transfer_completed"]`
+    /// * data - `[new_owner: Address]`
+    fn accept_ownership(e: &Env, caller: Address);
```

Comment:
Do we actually need the `caller` param here? It is supposed to be in the storage and can be fetched from there.

If you agree that we don't need it, that will affect the storage function in this module and `stellar_role_transfer::accept_transfer`.

Reply:
good catch, also changed the `accept_admin`